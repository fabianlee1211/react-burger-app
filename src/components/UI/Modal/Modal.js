import React, { useMemo } from 'react';
import Backdrop from '../Backdrop/Backdrop';
import classes from './Modal.css';

const Modal = props => {
  return useMemo(() => {
    return (
      <React.Fragment>
        <Backdrop show={props.show} clicked={props.modalClosed} />
        <div
          className={classes.Modal}
          style={{
            transform: props.show ? 'translateY(0)' : 'translateY(-100vh)',
            opacity: props.show ? '1' : '0'
          }}>
          {props.children}
        </div>
      </React.Fragment>
    );
  }, [props.children, props.show]);
};

export default Modal;
