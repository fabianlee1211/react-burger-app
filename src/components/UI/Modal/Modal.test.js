import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

describe('<Modal />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Modal />);
  });

  it('renders Modal correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a Backdrop and modal', () => {
    expect(wrapper.find('Backdrop')).toHaveLength(1);
    expect(wrapper.find('div.Modal')).toHaveLength(1);
  });

  it('renders modal children', () => {
    wrapper.setProps({ children: 'Test' });
    expect(wrapper.find('div.Modal').props().children).toBe('Test');
  });
});
