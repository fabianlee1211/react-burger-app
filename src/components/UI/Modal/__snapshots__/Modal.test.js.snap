// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders Modal correctly 1`] = `
<Fragment>
  <Backdrop />
  <div
    className="Modal"
    style={
      Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      }
    }
  />
</Fragment>
`;
